app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -58,11 +63,19 @@ import org.mozilla.fenix.tabstray.ext.collectionMessage
 import org.mozilla.fenix.tabstray.ext.make
 import org.mozilla.fenix.tabstray.ext.orDefault
 import org.mozilla.fenix.tabstray.ext.showWithTheme
+import org.mozilla.fenix.tabstray.syncedtabs.SyncButtonBinding
+import org.mozilla.fenix.tabstray.syncedtabs.SyncedTabsListItem
+import org.mozilla.fenix.tabstray.syncedtabs.toComposeList
+import org.mozilla.fenix.tabstray.syncedtabs.toSyncedTabsListItem
 import org.mozilla.fenix.utils.allowUndo
 import kotlin.math.max
 
 @Suppress("TooManyFunctions", "LargeClass")
-class TabsTrayFragment : AppCompatDialogFragment() {
+class TabsTrayFragment :
+    AppCompatDialogFragment(),
+    SyncedTabsView,
```

Comment:
Let's implement this in an "integration" class.

It's to ask when adding things to a root class: what does Synced Tabs have to do with the TabsTray**Fragment**? 

When we add more things to the fragment, we start creating objects that do too many things and making testing harder.